gremlin/ogma-neptune/README.md

```diff
@@ -0,0 +1,248 @@
+```
+
+Note the value of "id" field from the earlier output and use it as a `<rest-api-id>` value below.
+
+```
+aws apigateway get-resources --rest-api-id <rest-api-id>
+```
+
+Note the value of "id" field from the earlier output and use it as a `<parent-id>` value below.
+Below command would create a resource under the root strutrure of the API.
+
+```
+aws apigateway create-resource --rest-api-id <rest-api-id>  --parent-id <parent-id> --path-part {proxy+}
+```
+
+Note the value of "id" field from the output and use it as a `<resource-id>` in the command below.
+
+```
+aws apigateway put-method --rest-api-id <rest-api-id>  --resource-id <resource-id> --http-method ANY \
+--authorization-type NONE
+```
+
+So far we created, an API, API Resource and Methods for that Resource (GET/PUT/POST/DELETE or ANY for all methods).
+We will now create the API method integration, that would identify the AWS Lambda function for which this resource will be acting as a PROXY.
+
+Use appropriate values obtained from the previous commands.
+
+```
+aws apigateway put-integration --rest-api-id <rest-api-id> \
+--resource-id  <resource-id> --http-method ANY --type AWS_PROXY \
+--integration-http-method POST  \
+--uri arn:aws:apigateway:<aws-region-code>:lambda:path/2015-03-31/functions/arn:aws:lambda:<aws-region-code>:<aws-account-number>:function:<lambda-function-name>/invocations
+```
+
+Finally we deploy the API using below command.
+
+```
+aws apigateway create-deployment --rest-api-id <rest-api-id> --stage-name test
+```
+
+In order for Amazon API Gateway API, to invoke AWS Lambda function we either need to provide the "execution-role" to API Integration or we can also add the permission (subscription) in AWS Lambda explicitly that says an API "X" can invoke a lambda function. This API Gateway subscription is also reflected in AWS Console.
+
+Execute below command to add API Gateway subscription/permission to AWS Lambda function.
+
+```
+aws lambda add-permission --function-name <lambda-function-name> \
+--statement-id <any-unique-id> --action lambda:* \
+--principal apigateway.amazonaws.com \
+--source-arn arn:aws:execute-api:<aws-region-code>:<aws-account-number>:<rest-api-id>/*/*/*
+```
+
+We have now created an API Gateway proxy for the AWS Lambda function.
+
+### 5. Configure Amazon S3 bucket for hosting a static website
+
+Now that we have all the backend infrastructure ready for handling the API requests to get data out from Amazon Neptune, let's create an Amazon S3 bucket that will be used to host a static website.<br/>
+
+Run below commands to create an Amazon S3 bucket as a static website and upload the `visualize-graph.html` into it.
+
+```
+--create Amazon S3 bucket with public read access
+aws s3api create-bucket --bucket <bucket-name> --acl public-read --region <aws-region-code> --create-bucket-configuration LocationConstraint=<aws-region-code>
```

Comment:
I would suggest linking to this page [https://github.com/aws-samples/amazon-cloudfront-secure-static-site#user-content-amazon-cloudfront-secure-static-website](url) so the person reading this creates a CloudFront distribution rather than use S3 to host the static website.

First, create a CloudFront Origin Access Id using the following command:
`aws cloudfront create-cloud-front-origin-access-identity --cloud-front-origin-access-identity-config  CallerReference="linkurious example",Comment="linkurious example"`

Make a note of the value of the "Id" field.

Next, create a CloudFront distribution to the newly created S3 bucket, using the following command. 
**NOTE: Make sure to replace the `<bucket-name>` and `<origin-id>` placeholders with your own values.**
`aws cloudfront create-distribution --distribution-config file://cloudfront-create-distribution-config.txt`
[cloudfront-create-distribution-config.txt](https://github.com/aws-samples/amazon-neptune-samples/files/9149260/cloudfront-create-distribution-config.txt)

Make a note of the value of the `DomainName` field.